File: longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n=strs.size();
        if(n==0){
            return "";
        }
        string str="";
        sort(begin(strs),end(strs));
        string a=strs[0];
        string b=strs[n-1];
        for(int i=0;i<a.size();i++){
            if(a[i]==b[i]){
                str+=a[i];
            }else{
                break;
            }
        }
        return str;
    }
};

File: longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) {
            return "";
        }
        Arrays.sort(strs);
        String firstStr = strs[0];
        String lastStr = strs[strs.length - 1];
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < firstStr.length(); i++) {
            if (firstStr.charAt(i) == lastStr.charAt(i)) {
                sb.append(firstStr.charAt(i));
            } else {
                break;
            }
        }
        return sb.toString();
    }
}
